Issue #9 -Implement hook_config_info()

yorkshire-pudding opened the pull request:
Implement hook_config_info() so export of config possible in configuration manager

yorkshire-pudding committed:
Implement hook_config_info()

Implement hook_config_info() so export of config possible in configuration manager

alvinjohnsonso commented on tawk_to.module:
```diff
@@ -206,3 +206,15 @@ function tawk_to_remove_widget() {
+ * Implements hook_config_info().
+ */
+function tawk_to_config_info() {
+    $info[TawkHelper::TAWK_TO_CONFIG_NAME] = array(
```
@yorkshire-pudding is `$info` here always defined? Might be better to add a fail-safe here so that it won't cause any `undefined` error. Also, there are extra spaces at the beginning of the line.

alvinjohnsonso replied:
NVM on the `$info` part. Though, I did take a look at the documentation and some examples on the hook. I found that instead of using `$info`, they're using the `$prefixes` variable to set the configuration info. I think it would be better if we can also do the same so we can be uniform with the others.

Ref:
- https://docs.backdropcms.org/api/backdrop/core%21modules%21config%21config.api.php/function/hook_config_info/1
- https://docs.backdropcms.org/api/backdrop/core%21modules%21admin_bar%21admin_bar.module/function/admin_bar_config_info/1
- https://docs.backdropcms.org/api/backdrop/modules%21examples%21ajax_example%21ajax_example.module/function/ajax_example_config_info/1
- https://docs.backdropcms.org/api/backdrop/core%21modules%21block%21block.module/function/block_config_info/1
- https://docs.backdropcms.org/api/backdrop/modules%21examples%21action_example%21action_example.module/function/action_example_config_info/1

yorkshire-pudding replied:
Thanks for reviewing this @alvinjohnsonso - I've looked at the docs and you're right that in core modules they are using prefixes.  I was basing it on another contrib module that I'd recently found a config bug in, but happy to make it consistent with core.  I've corrected the spacing at the beginning of each line.

yorkshire-pudding committed:
Replace $info with $prefixes, fix spacing

Replace $info with $prefixes to be consistent with core modules and fixed spacing at beginning of lines

alvinjohnsonso commented:
@yorkshire-pudding would you happen to know if we're able to provide additional details to the exported configuration? We were thinking if we can add the plugin's version to the exported configuration.

yorkshire-pudding commented:
> @yorkshire-pudding would you happen to know if we're able to provide additional details to the exported configuration? We were thinking if we can add the plugin's version to the exported configuration.

We set the config in tawk_to.admin.inc, though there is no need for someone to update config just when they update the module.  In theory, we could look at adding elsewhere so it gets updated, but that is a potential overhead as it would write to the config file each time.

As config export is really for ensuring we can move config from one site to another (e.g. dev/stage to production), I'm not clear on what the benefits of exporting the module version would be. That would either be updated on each instance or updated via version control.

If the config function were to add project versions at run time, it would need a change to core.
